packages/cli/yaml/validator/src/ComplexTypeDetector.ts

```diff
@@ -12,7 +14,8 @@ import {
     isRawDiscriminatedUnionDefinition,
     isRawEnumDefinition,
     isRawObjectDefinition,
-    isRawUndiscriminatedUnionDefinition
+    isRawUndiscriminatedUnionDefinition,
+    RawSchemas
```

Comment:
Just want to make sure that the changes to `ComplexTypeDetector` won't affect something else in an unintended way

Reply:
Main changes are:
* `isTypeComplex` returns false for  _simple_ objects, such that they don't define any [nested] literal or union properties. In other words, we consider an object "complex" if it contains any literal or union properties.
* The `visited` map is required to prevent infinite loops for recursive types.

I can also add a comment on the `isTypeComplex` function if that would help. I considered doing so but left it out at first.

Reply:
`ComplexTypeDetector` is _only_ used in the query parameter validator for now.

Reply:
ah amazing -- surprised its not used by path params and headers

Reply:
can we rename to `ComplexQueryParamTypeDetector`